### ISSUE TITLE
Firefox: Short read; Could not decrypt record or bad MAC

### ISSUE THREAD

**Issue by rentallect** [openziti/ziti-sdk-js#31]
Seeing different behavior depending on which browser is used (with `ziti-sdk-js` 0.6.3+)

Although the `tmdb` web-app test-mule runs very solidly under `Chrome (87.0.4280.67)`, it intermittently fails under `Firefox (83.0)` on a Macbook.

The problem manifests on the edge router as:

`ERROR [foundation/channel2.(*channelImpl).rxer [ch{edge}->u{classic}->i{ORXE}] rx error (short read)`

The edge router is attempting to read the 4-byte magic number for the edge-protocol message, but gets zero bytes instead. The error is unrecoverable on edge router side, so it closes the channel.

The client side then gets confused, and the attempt to decrypt incoming traffic/responses, manifests an error:

`Could not decrypt record or bad MAC`

There seems to be some kind of async/timing differences, or websocket data-chunking differences, concerning how things operate on these two browsers while running the exact same code.

**Comment by smilindave26** [comment]
Could this be the e2e=Off issue with ‘old’ edge routers?

**Comment by rentallect** [comment]
@smilindave26 No, not in this scenario, because I'm running these tests against a locally-deployed Ziti build/network, that contains `realease-next + my-forge-poc` stuff.  

fwiw, the failure behavior doesn't change if I swap the service from encryption ON or OFF.  Always works with Chrome either way, tends to fail with FF either way.

Current evidence suggests that `ziti-sdk-js` executes slightly differently depending on the browser that is hosting it. i.e. Safari has a slightly different issue which I will open once I have more relevant info.
